File: apps/mobile/ios/Uniswap/Widget/RNWidgets.m
//
//  RNWidgets.m
//  Uniswap
//
//  Created by Eric Huang on 8/2/23.
//

#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(RNWidgets, NSObject)

RCT_EXTERN_METHOD(hasWidgetsInstalled: (RCTPromiseResolveBlock *)resolve
                  reject:(RCTPromiseRejectBlock *)reject)

@end

File: apps/mobile/ios/Uniswap/Widget/RNWidgets.swift
//
//  RNWidgets.swift
//  Uniswap
//
//  Created by Eric Huang on 8/2/23.
//

import Foundation
import WidgetKit

@objc(RNWidgets)
class RNWidgets: NSObject {
  
  @objc
  func hasWidgetsInstalled(_ resolve: @escaping RCTPromiseResolveBlock, reject: @escaping RCTPromiseRejectBlock) {
    WidgetCenter.shared.getCurrentConfigurations() { result in
      if case .success(let config) = result {
        resolve(config.count > 0)
      } else {
        resolve(false)
      }
    }
  }
  
  @objc
  static func requiresMainQueueSetup() -> Bool {
    return false
  }
}
